File: exercise_2/app/src/main/java/com/example/exercise_2/QuizActivity.java
package com.example.exercise_2;

import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

public class QuizActivity extends AppCompatActivity {
    Button answer1, answer2, answer3, answer4;

    TextView score, question;

    private Questions mQuestions = new Questions();

    int r;

    private String mAnswer;
    public int mScore = 0;
    private int mQuestionLength = mQuestions.mQuestions.length;

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_quiz);

        r = 0;

        answer1 = (Button) findViewById(R.id.answer1);
        answer2 = (Button) findViewById(R.id.answer2);
        answer3 = (Button) findViewById(R.id.answer3);
        answer4 = (Button) findViewById(R.id.answer4);

        score = (TextView) findViewById(R.id.score);
        question = (TextView) findViewById(R.id.question);

        score.setText("Wynik: " + mScore);

        updateQuestion(r);

        answer1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (answer1.getText() == mAnswer) {
                    mScore++;
                    score.setText("Wynik : " + mScore);
                    r++;
                    updateQuestion(r);
                } else {
                    gameOver();
                }
            }
        });

        answer2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (answer2.getText() == mAnswer) {
                    mScore++;
                    score.setText("Wynik : " + mScore);
                    r++;
                    updateQuestion(r);
                } else {
                    gameOver();
                }
            }
        });

        answer3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (answer3.getText() == mAnswer) {
                    mScore++;
                    score.setText("Wynik : " + mScore);
                    r++;
                    updateQuestion(r);
                } else {
                    gameOver();
                }
            }
        });

        answer4.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (answer4.getText() == mAnswer) {
                    mScore++;
                    score.setText("Wynik : " + mScore);
                    r++;
                    updateQuestion(r);
                } else {
                    gameOver();
                }
            }
        });
    }

    private void updateQuestion(int num) {
        if (num > 5) {
            endTheGame();
        } else {
            question.setText(mQuestions.getQuestion(num));
            answer1.setText(mQuestions.getChoice1(num));
            answer2.setText(mQuestions.getChoice2(num));
            answer3.setText(mQuestions.getChoice3(num));
            answer4.setText(mQuestions.getChoice4(num));

            mAnswer = mQuestions.getCorrectAnwer(num);
        }
    }

    private void gameOver() {
        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(QuizActivity.this);
        alertDialogBuilder
                .setMessage("Zła odpowiedz udało Ci się uzyskać  " + mScore + " " + "punktów.")
                .setCancelable(false)
                .setPositiveButton("Zacznij od nowa",
                        new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialogInterface, int i) {
                                startActivity(new Intent(getApplicationContext(), QuizActivity.class));
                                finish();
                            }
                        })

                .setNegativeButton("Wyjscie",
                        new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialogInterface, int i) {
                                Toast.makeText(QuizActivity.this, "", Toast.LENGTH_LONG).show();
                                finish();
                            }
                        });

        AlertDialog alertDialog = alertDialogBuilder.create();
        alertDialog.show();
    }

    private void endTheGame() {
        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(QuizActivity.this);
        alertDialogBuilder
                .setMessage("Gratulacje uzyskałeś  " + mScore + " " + "punktów.")
                .setCancelable(false)
                .setPositiveButton("Zacznij od nowa",
                        new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialogInterface, int i) {
                                startActivity(new Intent(getApplicationContext(), QuizActivity.class));
                                finish();
                            }
                        })

                .setNegativeButton("Wyjscie",
                        new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialogInterface, int i) {
                                Toast.makeText(QuizActivity.this, "", Toast.LENGTH_LONG).show();
                                finish();
                            }
                        });

        AlertDialog alertDialog = alertDialogBuilder.create();
        alertDialog.show();
    }
}

File: BMI_Kotlin/app/src/main/java/com/example/bmi_kotlin/QuizActivity.kt
package com.example.bmi_kotlin

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AlertDialog
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.TextView
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_quiz.*

class QuizActivity : AppCompatActivity() {
    private val mQuestions = Questions()
    private var mAnswer: String? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_quiz)
        var r: Int = 0
         var mScore: Int = 0
        score.setText("Wynik: $mScore")
        updateQuestion(r)

        answer1.setOnClickListener {
            if (answer1.text === mAnswer) {
                mScore++
                score.text = "Wynik : $mScore"
                r++
                updateQuestion(r)
            } else {
                gameOver(mScore)
            }
        }

        answer2.setOnClickListener {
            if (answer2.text === mAnswer) {
                mScore++
                score.text = "Wynik : $mScore"
                r++
                updateQuestion(r)
            } else {
                gameOver(mScore)
            }
        }

        answer3.setOnClickListener {
            if (answer3.text === mAnswer) {
                mScore++
                score.text = "Wynik : $mScore"
                r++
                updateQuestion(r)
            } else {
                gameOver(mScore)
            }
        }

        answer4.setOnClickListener {
            if (answer4.text === mAnswer) {
                mScore++
                score.text = "Wynik : $mScore"
                r++
                updateQuestion(r)
            } else {
                gameOver(mScore)
            }
        }
    }

    private fun updateQuestion(num: Int) {
        if (num > 5) {
            endTheGame(num)
        } else {
            question.setText(mQuestions.getQuestion(num))
            answer1.setText(mQuestions.getChoice1(num))
            answer2.setText(mQuestions.getChoice2(num))
            answer3.setText(mQuestions.getChoice3(num))
            answer4.setText(mQuestions.getChoice4(num))
            mAnswer = mQuestions.getCorrectAnwer(num)
        }
    }

    private fun gameOver(mScore:Int) {
        val alertDialogBuilder =
            AlertDialog.Builder(this@QuizActivity)
        alertDialogBuilder
            .setMessage("Zła odpowiedz udało Ci się uzyskać  $mScore punktów.")
            .setCancelable(false)
            .setPositiveButton(
                "Zacznij od nowa"
            ) { dialogInterface, i ->
                startActivity(Intent(applicationContext, QuizActivity::class.java))
                finish()
            }
            .setNegativeButton(
                "Wyjscie"
            ) { dialogInterface, i ->
                Toast.makeText(this@QuizActivity, "", Toast.LENGTH_LONG).show()
                finish()
            }
        val alertDialog = alertDialogBuilder.create()
        alertDialog.show()
    }

    private fun endTheGame(mScore:Int) {
        val alertDialogBuilder =
            AlertDialog.Builder(this@QuizActivity)
        alertDialogBuilder
            .setMessage("Gratulacje uzyskałeś  $mScore punktów.")
            .setCancelable(false)
            .setPositiveButton(
                "Zacznij od nowa"
            ) { dialogInterface, i ->
                startActivity(Intent(applicationContext, QuizActivity::class.java))
                finish()
            }
            .setNegativeButton(
                "Wyjscie"
            ) { dialogInterface, i ->
                Toast.makeText(this@QuizActivity, "", Toast.LENGTH_LONG).show()
                finish()
            }
        val alertDialog = alertDialogBuilder.create()
        alertDialog.show()
    }


}